docs/core/diagnostics/snippets/lifetime-health-checks/Program.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Hosting;
+
+var builder = Host.CreateApplicationBuilder(args);
+
+var healthChecksBuilder = builder.Services
+    .AddHealthChecks()
+    .AddApplicationLifecycleHealthCheck();
+
+var app = builder.Build();
+
+var lifetime = app.Services.GetRequiredService<IHostApplicationLifetime>();
+var healthCheckService = app.Services.GetRequiredService<HealthCheckService>();
+
+void ReportHealthCheck(string eventName)
+{
+    // ⚠️ Normally, you would await the result of CheckHealthAsync, but this
```

Comment:
Can we show correct code instead? For instance using startup initialization (if it is not removed from extensions yet), or creating hosted service (.NET 8 hosted services are associated with lifetime - https://source.dot.net/#Microsoft.Extensions.Hosting.Abstractions/IHostedLifecycleService.cs,7c4b39d9d078193f).

Reply:
I rewrote this to implement the `IHostedLifecylceService`, and for all six events - it reports unhealthy. Is that really what you think we should show?